gateway/scim.py

```diff
@@ -66,95 +66,20 @@ def scim_addusers():
-    pairing_just_works_key = None
-    pairing_pass_key = None
-    pairing_oob_key = None
-    pairing_oobrn = None
-
-    endpoint_apps = None
-    endpoint_apps_ext = request.json.get(
-        "urn:ietf:params:scim:schemas:extension:endpointAppsExt:2.0:Device", None)
-    if endpoint_apps_ext is not None:
-        applications = endpoint_apps_ext.get("applications")
-        endpoint_app_ids = [app.get("value") for app in applications]
-        # Select all endpoint apps from the database
-        endpoint_apps = session.scalars(select(EndpointApp).filter(
-            EndpointApp.id.in_(endpoint_app_ids))).all()
-
-    if pairing_just_works:
-        pairing_just_works_key = pairing_just_works.get("key")
-    pairing_pass = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "urn:ietf:params:scim:schemas:extension:pairingPassKey:2.0:Device")
-    if pairing_pass:
-        pairing_pass_key = pairing_pass.get("key")
-    pairing = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "urn:ietf:params:scim:schemas:extension:pairingOOB:2.0:Device")
-    if pairing:
-        pairing_oob_key = pairing.get("key")
-        pairing_oobrn = pairing.get("randNumber")
-    device_id = request.json.get("id")
-
-    existing_device = Device.query.filter_by(
-        device_mac_address=device_mac_address).first()
-
-    if existing_device:
-        return make_response(
-            jsonify(
-                {
-                    "schemas": ["urn:ietf:params:scim:schemas:core:2.0:Device"],
-                    "detail": "User already exists in the database.",
-                    "status": 409,
-                }
-            ),
-            409,
-        )
-
-    try:
-        user = Device(
-            device_id=device_id,
-            schemas=schemas,
-            device_display_name=device_display_name,
-            admin_state=admin_state,
-            version_support=version_support,
-            device_mac_address=device_mac_address,
-            is_random=is_random,
-            separate_broadcast_address=separate_broadcast_address,
-            irk=irk,
-            pairing_methods=pairing_methods,
-            pairing_null=pairing_null,
-            pairing_just_works_keys=pairing_just_works_key,
-            pairing_pass_key=pairing_pass_key,
-            pairing_oob_key=pairing_oob_key,
-            pairing_oobrn=pairing_oobrn,
-            endpoint_apps=endpoint_apps,
-            created_time=datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ"),
-        )
-        session.add(user)
-
-        session.commit()
-        return make_response(jsonify(user.serialize()), 201)
-    except Exception as e:
-        return str(e)
 
+    # Dispatch to appropriate function
+    if 'urn:ietf:params:scim:schemas:extension:ble:2.0:Device' in schemas:
+        return ble_create_device(request)
+    return make_response(
```

Comment:
Looks like the spec allows creation of a device without any extension (see [here](https://github.com/iot-onboarding/scim-devices/blob/main/examples/SCIM_device_core_schema_object.json)). Not sure what use it would be but it allows it.

Reply:
I think it's safe to say that we should fix that in the draft ;-). However, we probably need to support **more** than one extension.  And as soon as I have the wifi code, we'll be well on the way to that ;-)

Reply:
@rohitmohan96 can you pull this branch and give it a quick test?

Reply:
@rohitmohan96 please review and approve